src/domain/youthProfile/create/CreateYouthProfile.tsx

```diff
@@ -70,6 +71,8 @@ function CreateYouthProfile({
   >(UPDATE_PROFILE);
 
   const birthDate = getCookie('birthDate');
+  const isBirthDateValid = isValid(parseISO(birthDate));
+  if (!isBirthDateValid) history.replace('/login');
```

Comment:
Oh yea, I was supposed to combine those but forgot (tests crashing redirected my attention). 

But on another note. I'm not sure how user's managed to cause this error in the first place. My guess is that maybe the previous versions of `BirthDateInput` haven't caught all the possible errors. This is of course now fixed as well, but to make sure this error will not occur again, I wan't to make it as bulletproof as possible. 

It also seems that if all cookies are disabled, authentication won't succeed because all login methods provided by tunnistamo require cookies to login.

Reply:
In that case I wonder if we should not use the validation function `date-fn` provides in the `BirthDate` input? I think that currently it just uses `yup`'s default `date()` schema which is likely just trying to cast the date into a JS `Date` object.

From yup's readme:
>The default cast logic of date is pass the value to the Date constructor, failing that, it will attempt to parse the date as an ISO date string.

>Failed casts return an invalid Date.

I don't entirely understand how an incorrect date would end up passing this validation, but it wouldn't surprise me if there's a way.

Reply:
It could be one option, not necessary for now at least. The newest error handling you added are not yet on production, and what I understand and tested it, should catch all invalid dates. 

Even if the `BirthDateInput` validates dates 100% correct, user can still manually tamper the cookies, which will result into error if value is invalid.